File: Backend/src/main/java/dev/rohitrai/Finance/model/Transaction.java
package dev.rohitrai.Finance.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@AllArgsConstructor
@Builder
@Entity
@Getter
@NoArgsConstructor
@Setter
public class Transaction {
    @Id
    private UUID id;
    private double amount;
    private TransactionType type;
    private String description;
    private String tags;
    private long date;
    @CreationTimestamp
    private long createTime;
    @UpdateTimestamp
    private long updateTime;
}

File: Android/app/src/main/java/com/example/finance/data/Transaction.kt
package com.example.finance.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    val amount: Double,
    val type: TransactionType,
    val description: String,
    val tags: String,
    val date: Date,
    @ColumnInfo(name = "create_time")
    val createTime: Date,
    @ColumnInfo(name = "update_time")
    val updateTime: Date,
)
